pkg/sync/syncWorker.go

```diff
@@ -430,10 +452,11 @@ func (s *syncWorker) validateAndInsertEnvelope(
 		}
 	}()
 
-	env, err := envUtils.NewOriginatorEnvelope(envProto)
+	var env *envUtils.OriginatorEnvelope
+	env, err = envUtils.NewOriginatorEnvelope(envProto)
 	if err != nil {
 		s.log.Error("Failed to unmarshal originator envelope", zap.Error(err))
-		return err
+		return backoff.Permanent(err)
```

Comment:
this will shut down the worker. Which, given how complex this nesting is, will retry again here ```tracing.GoPanicWrap(
		s.ctx,
		&s.wg,
		fmt.Sprintf("node-subscribe-%d", nodeid),
		func(ctx context.Context) {
			for {
				select {
				case <-ctx.Done():
					return
				default:
					config := s.setupNodeRegistration(ctx, nodeid)
					s.subscribeToNodeRegistration(config)
				}
			}
		})```

Reply:
what should we do when we receive something we cant process?